test/e2e/proxy_test.go

```diff
@@ -68,7 +70,493 @@ func skipIfProxyDisabled(t *testing.T) {
+	}
+
+	createTestAgnhostPods(t, data, testPodNameCp, testPodNameWk, false)
+	portCluster, portLocal := createNodePortServices(t, data, isIPv6)
+	clusterUrlCp := net.JoinHostPort(nodeCpIP, portCluster)
+	clusterUrlWk := net.JoinHostPort(nodeWkIP, portCluster)
+	clusterUrlLo := net.JoinHostPort(localhostIP, portCluster)
+	localUrlCp := net.JoinHostPort(nodeCpIP, portLocal)
+	localUrlWk := net.JoinHostPort(nodeWkIP, portLocal)
+	localUrlLo := net.JoinHostPort(localhostIP, portLocal)
+
+	t.Run("Pod CIDR Endpoints", func(t *testing.T) {
+		nodePortTestCases(t, data, clusterUrlCp, clusterUrlWk, clusterUrlLo, localUrlCp, localUrlWk, localUrlLo,
+			nodeCpIP, nodeWkIP, podClientCpIP, podClientWkIP, testPodNameCp, testPodNameWk, clientPodCp, clientPodWk, false)
+	})
+
+	testPodHostNetworkNameCp := "agnhost-cp-h"
+	testPodHostNetworkNameWk := "agnhost-wk-h"
+	nodeNameCp := controlPlaneNodeName()
+	nodeNameWk := workerNodeName(1)
+
+	deleteTestAgnhostPods(t, data, testPodNameCp, testPodNameWk)
+	createTestAgnhostPods(t, data, testPodHostNetworkNameCp, testPodHostNetworkNameWk, true)
+	t.Run("Host Network Endpoints", func(t *testing.T) {
+		nodePortTestCases(t, data, clusterUrlCp, clusterUrlWk, clusterUrlLo, localUrlCp, localUrlWk, localUrlLo,
+			nodeCpIP, nodeWkIP, podClientCpIP, podClientWkIP, nodeNameCp, nodeNameWk, clientPodCp, clientPodWk, true)
+	})
+}
+
+func nodePortTestCases(t *testing.T, data *TestData, clusterUrlCp, clusterUrlWk, clusterUrlLo, localUrlCp, localUrlWk, localUrlLo string,
+	nodeCpIP, nodeWkIP, podClientCpIP, podClientWkIP, testPodHostnameCp, testPodHostnameWk, clientPodCp, clientPodWk string, hostNetwork bool) {
+	t.Run("Case=ExternalTrafficPolicy:Cluster Client:Remote", func(t *testing.T) {
+		testNodePortClusterFromRemote(t, clusterUrlCp, clusterUrlCp)
+	})
+	t.Run("Case=ExternalTrafficPolicy:Cluster Client:Local", func(t *testing.T) {
+		testNodePortClusterFromLocal(t, data, clusterUrlCp, clusterUrlWk, clusterUrlLo)
+	})
+	t.Run("Case=ExternalTrafficPolicy:Cluster Client:Pod", func(t *testing.T) {
+		testNodePortClusterFromPod(t, data, clusterUrlCp, clusterUrlCp, clientPodCp, clientPodWk)
+	})
+	t.Run("Case=ExternalTrafficPolicy:Local Client:Remote", func(t *testing.T) {
+		if hostNetwork {
+			t.Skipf("Skip this test if Endpoint is on host network")
+		}
+		testNodePortLocalFromRemote(t, localUrlCp, localUrlWk, nodeCpIP, nodeWkIP, testPodHostnameCp, testPodHostnameWk)
+	})
+	t.Run("Case=ExternalTrafficPolicy:Local Client:Local", func(t *testing.T) {
+		testNodePortLocalFromLocal(t, data, localUrlCp, localUrlWk, localUrlLo, testPodHostnameCp, testPodHostnameWk)
+	})
+	t.Run("Case=ExternalTrafficPolicy:Local Client:Pod", func(t *testing.T) {
+		testNodePortLocalFromPod(t, data, localUrlCp, localUrlWk, clientPodCp, clientPodWk, podClientCpIP, podClientWkIP, testPodHostnameCp, testPodHostnameWk)
+	})
+}
+
+func createTestAgnhostPods(t *testing.T, data *TestData, agnhostCp, agnhostWk string, hostNetwork bool) {
```

Comment:
This has same problem of redundancy as others. Can we make it generic by just creating one Pod and calling it twice? Otherwise people will have to create another function when they also want to create agnhost Pod. Maybe it's already redundant with existing function

Reply:
Updated.